ogr/ogrsf_frmts/miramon/mm_gdal_functions.c

```diff
@@ -0,0 +1,2801 @@
+
+    if (cal_DBF_estesa || bd_xp->nFields > MM_MAX_N_CAMPS_DBF_CLASSICA ||
+        bd_xp->nRecords > UINT32_MAX)
+        bd_xp->dbf_version = (MM_BYTE)MM_MARCA_VERSIO_1_DBF_ESTESA;
+    else
+        bd_xp->dbf_version = MM_MARCA_DBASE4;
+}
+
+static void
+MM_InitializeOffsetExtendedFieldNameFields(struct MM_DATA_BASE_XP *bd_xp,
+                                           MM_EXT_DBF_N_FIELDS i_camp)
+{
+    memset((char *)(&bd_xp->pField[i_camp].reserved_2) +
+               MM_OFFSET_RESERVAT2_OFFSET_NOM_ESTES,
+           0, 4);
+}
+
+static void
+MM_InitializeBytesExtendedFieldNameFields(struct MM_DATA_BASE_XP *bd_xp,
+                                          MM_EXT_DBF_N_FIELDS i_camp)
+{
+    memset((char *)(&bd_xp->pField[i_camp].reserved_2) +
+               MM_OFFSET_RESERVED2_EXTENDED_NAME_SIZE,
+           0, 1);
+}
+
+static short int MM_return_common_valid_DBF_field_name_string(char *cadena)
+{
+    char *p;
+    short int error_retornat = 0;
+
+    if (!cadena)
+        return 0;
+    //strupr(cadena);
+    for (p = cadena; *p; p++)
+    {
+        (*p) = (char)toupper(*p);
+        if ((*p >= 'A' && *p <= 'Z') || (*p >= '0' && *p <= '9') || *p == '_')
+            ;
+        else
+        {
+            *p = '_';
+            error_retornat |= MM_FIELD_NAME_CHARACTER_INVALID;
+        }
+    }
+    if (cadena[0] == '_')
+    {
+        cadena[0] = '0';
+        error_retornat |= MM_FIELD_NAME_FIRST_CHARACTER_;
+    }
+    return error_retornat;
+}
+
+static short int MM_ReturnValidClassicDBFFieldName(char *cadena)
+{
+    size_t long_nom_camp;
+    short int error_retornat = 0;
+
+    long_nom_camp = strlen(cadena);
+    if ((long_nom_camp < 1) ||
+        (long_nom_camp >= MM_MAX_LON_CLASSICAL_FIELD_NAME_DBF))
+    {
+        cadena[MM_MAX_LON_FIELD_NAME_DBF - 1] = '\0';
+        error_retornat |= MM_FIELD_NAME_TOO_LONG;
+    }
+    error_retornat |= MM_return_common_valid_DBF_field_name_string(cadena);
+    return error_retornat;
+}
+
+static MM_BOOLEAN
+MM_CheckClassicFieldNameEqual(const struct MM_DATA_BASE_XP *data_base_XP,
+                              const char *classical_name)
+{
+    MM_EXT_DBF_N_FIELDS i;
+
+    for (i = 0; i < data_base_XP->nFields; i++)
+    {
+        if ((strcasecmp(data_base_XP->pField[i].ClassicalDBFFieldName,
+                        classical_name)) == 0 ||
+            (strcasecmp(data_base_XP->pField[i].FieldName, classical_name)) ==
+                0)
+            return TRUE;
+    }
+    return FALSE;
+}
+
+static char *MM_GiveNewStringWithCharacterInFront(const char *text,
+                                                  char caracter)
+{
+    char *ptr;
+    size_t i;
+
+    if (!text)
+        return nullptr;
+
+    i = strlen(text);
+    if ((ptr = calloc_function(i + 2)) == nullptr)
+        return nullptr;
+
+    *ptr = caracter;
+    memcpy(ptr + 1, text, i + 1);
+    return ptr;
+}
+
+static char *MM_SetSubIndexFieldNam(char *nom_camp, MM_EXT_DBF_N_FIELDS index,
```

Comment:
If nom_camp is not modified by this function:

```suggestion
static char *MM_SetSubIndexFieldNam(const char *nom_camp, MM_EXT_DBF_N_FIELDS index,
```

Reply:
done!